src/webhook/upgrade_operator/fetch_upgradeconfig.py

```diff
@@ -0,0 +1,20 @@
+import os
+from kubernetes import client, config
+from kubernetes.client.rest import ApiException
+
+# Use KUBECONFIG from pod else consume from local ~/.kube/config
+if 'KUBERNETES_PORT' in os.environ:
```

Comment:
It would be more pythonic (and only a *teeny* bit less efficient) to do this like:

```python
try:
    config.load_incluster_config()
except ConfigException:
    config.load_kube_config()
```

Reply:
Also, it’s concerning that you’re doing this at the file level, meaning that it’ll happen at import time. Are you sure that’s what you want to do?

Reply:
Agreed. Actually, I was thinking whether to include the `config.load_kube_config()` or not as the client authentication would always happen using `config.load_incluster_config()` (from within the pod). But while testing the script this would be helpful when connected to a cluster remotely using `oc` client so included it anyway.

Reply:
Have added `ConfigException` to check for kube auth